framework/areg/base/private/Version.cpp

```diff
@@ -87,15 +89,24 @@ Version & Version::convFromString( const char * version )
     mMinor  = 0;
     mPatch  = 0;
 
-    String temp(version), major, minor, patch;
-    NEString::CharPos pos = NEString::START_POS;
-    pos = temp.substring( major, NECommon::OBJECT_SEPARATOR, pos );
-    pos = temp.substring( minor, NECommon::OBJECT_SEPARATOR, pos );
-    pos = temp.substring( patch, NECommon::OBJECT_SEPARATOR, pos );
+    std::string temp = version;
+
+    auto pos1 = temp.find_first_of(NECommon::OBJECT_SEPARATOR);
+    if (pos1 != std::string::npos)
+    {
+        const std::string major = temp.substr(0, pos1);
+        mMajor = std::stoi(major);
+    }
 
-    mMajor  = major.convToUInt32();;
-    mMinor  = minor.convToUInt32();
-    mPatch  = (mMajor != 0) && (mMinor != 0) ? patch.convToUInt32() : 0;
+    auto pos2 = temp.find_last_of(NECommon::OBJECT_SEPARATOR);
+    if (pos2 != std::string::npos && pos2 != pos1)
+    {
+        const std::string minor = temp.substr(pos1 + 1, pos2 - pos1 - 1);
+        mMinor = std::stoi(minor);
+
+        const std::string patch = temp.substr(pos2+1);
+        mPatch  = (mMajor != 0) && (mMinor != 0) ? std::stoi(patch) : 0;
+    }
```

Comment:
@biwiki
please double check the possible errors like:
"1", "1.", "1.9", "1.9.", ".9", ".9.", "1.2.3.4.5", etc. How the version is converted here?

Reply:
I thought about using std::regex here, and now I really think that I should go for it

Reply:
@biwiki, not sure about regex. Let's keep as simple as possible and use as less resources as possible. All what we do we'll need to adapt to RTOS. Even for C++17 not sure, because some RTOS support only C++11.